### DISCUSSION TITLE
Composer in prod env shows error on Slugger component

### DISCUSSION THREAD

**Discussion by spiider8** [symfony/symfony#49115] [Q&A]
Hi, i have problem with symfony slugger. When i run composer install --no-dev a then i clear cache, composer says:  You cannot use the "slugger" service since the Translation contracts are not installed.
But i dont want in my project(in production) translations library(with others dependecies....).
I looked to composer.json in symfony/string component and there is only dev dependecy on symfony/translation-contracts.

Can somebody help me pls?

**Comment by MatTheCat** [comment]
Hi, this is because the default `slugger` service is an instance of `AsciiSlugger` which implements `LocaleAwareInterface` from [symfony/translation-contracts](https://github.com/symfony/translation-contracts), so you have to require it.

Don’t worry though, it is different from [symfony/translation](https://github.com/symfony/translation) (it only defines three interfaces and a trait).

**Reply by spiider8** [reply]
Hi, thx for the answer. Ok but i will have to install [symfony/translation-contracts](https://github.com/symfony/translation-contracts) manually in my composer for now, but i think that it should be as dependecy in symfony/string component in require section and not only in require-dev section.....or not? Thanks so much

**Reply by MatTheCat** [reply]
You could require the string component without using the `AsciiSlugger`, that’s why it is your responsibility to provide the dependencies of what you’ll be using. Fortunately error messages are rather well-written!

**Reply by spiider8** [reply]
And can u help me how to do it pls?

**Reply by MatTheCat** [reply]
Uuuh it seems you’re using the `AsciiSlugger` so installing `symfony/translation-contracts` is all you need:

```shell
composer require symfony/translation-contracts
```

**Reply by spiider8** [reply]
I dont use AsciiSlugger, iam using only slugger:
`use Symfony\Component\String\Slugger\SluggerInterface;` 

but i think i can not install only slugger without AsciiSlugger by composer

**Reply by MatTheCat** [reply]
The `slugger` service is `AsciiSlugger` by default because it’s the only implementation of `SluggerInterface`. So you’re indeed using it.

**Reply by spiider8** [reply]
Good, thanks a lot for your answers
